packages/docgen/test/get-type-annotation.js

```diff
@@ -140,69 +147,109 @@ describe( 'Type annotations', () => {
 				"import( 'react' ).bar.baz.types.ComponentType[ 'displayName' ]"
 			);
 		} );
 
 		it( 'should get the parameterized tuple rest type', () => {
 			expect(
-				getTypeAnnotation( { ...paramTag, name: 'rest' }, node )
+				getTypeAnnotation( { ...paramTag, name: 'rest' }, node, 1 )
 			).toBe( '[ string | number, ...( keyof constant ) ]' );
 		} );
 
 		it( 'should get the type predicate return type', () => {
-			expect( getTypeAnnotation( returnTag, node ) ).toBe(
+			expect( getTypeAnnotation( returnTag, node, 1 ) ).toBe(
 				'foo is string'
 			);
 		} );
 	} );
 
-	describe( 'missing types', () => {
+	describe.skip( 'missing types', () => {
 		const node = getMissingTypesNode();
 
 		it( 'should throw an error if there is no return type', () => {
-			expect( () => getTypeAnnotation( returnTag, node ) ).toThrow(
+			expect( () => getTypeAnnotation( returnTag, node, 0 ) ).toThrow(
 				"Could not find return type for function 'fn'."
 			);
 		} );
 
 		it( 'should throw an error if there is no param type', () => {
-			expect( () => getTypeAnnotation( paramTag, node ) ).toThrow(
+			expect( () => getTypeAnnotation( paramTag, node, 0 ) ).toThrow(
 				"Could not find type for parameter 'foo' in function 'fn'."
 			);
 		} );
+	} );
 
-		it( 'should throw an error if it cannot find the param by name', () => {
-			expect( () =>
-				getTypeAnnotation( { ...paramTag, name: 'notFoo' }, node )
-			).toThrow(
-				"Could not find corresponding parameter token for documented parameter 'notFoo' in function 'fn'."
-			);
+	describe( 'function argument array-destructuring', () => {
+		describe( 'array-type', () => {
+			const node = getArrayDestructuringArrayTypeNode();
+
+			it( 'should grab the whole type for the unqualified name', () => {
+				expect( getTypeAnnotation( paramTag, node, 0 ) ).toBe( 'T[]' );
+			} );
+
+			it( 'should get the individual type for the qualified name', () => {
+				expect(
+					getTypeAnnotation( { ...paramTag, name: 'foo.0' }, node, 0 )
+				).toBe( 'T' );
+			} );
+		} );
+
+		describe( 'tuple-type', () => {
+			const node = getArrayDestructuringTupleTypeNode();
+
+			it( 'should grab the whole type for the unqualified name', () => {
+				expect( getTypeAnnotation( paramTag, node, 0 ) ).toBe(
+					'[ T, S ]'
+				);
+			} );
+
+			it( 'should get the individual type for the qualified name', () => {
+				expect(
+					getTypeAnnotation( { ...paramTag, name: 'foo.1' }, node, 0 )
+				).toBe( 'S' );
+			} );
+		} );
+
+		describe( 'any-other-type', () => {
+			const node = getArrayDestructuringAnyOtherTypeNode();
+
+			it( 'should get the full type name for the unqualified name', () => {
+				expect( getTypeAnnotation( paramTag, node, 0 ) ).toBe(
+					'( T & S ) | V'
+				);
+			} );
+
+			it( 'should get the full type name for the qualified name', () => {
+				expect(
+					getTypeAnnotation( { ...paramTag, name: 'foo.1' }, node, 0 )
+				).toBe( '( T & S ) | V' );
```

Comment:
like we discussed earlier, i would this example end up being `'( T & S ) | V.1'`?

Reply:
No, it should just be the element type of the array, which is `(T&S)|V`. The array was typed `(T&S)|V[]` so all the elements will have the same type.

Reply:
Oh wait shoot, I'm totally wrong, you're correct! It should be this:

`( ( T & S ) | V )[ 1 ]`

Reply:
ahh right, `T & S` also aren't index types